Library/Homebrew/formula.rb

```diff
@@ -2429,10 +2430,45 @@ def to_hash_with_variations
       end
     end
 
-    hash["variations"] = variations
+    hash["variations"] = @variations_hash
     hash
   end
 
+  # Builds internal json v3 hash.
+  #
+  # @private
+  def to_api_hash
+    api_hash = to_hash_with_variations.except(
+      # Included in the top-level of the `homebrew-core.json` file
+      # created by the `brew generate-formula-api` command.
+      "aliases",
+      "oldname",
+      "oldnames",
+      "tap",
+      "tap_git_head",
```

Comment:
- Could remove `options`, nothing in homebrew/core has them.
- Can remove `linked_keg`, `installed`, `pinned`, `outdated` as it's a reflection of the machine that generated this output

Reply:
I believe those will all be omitted automatically by the `api_has.reject { |_, value| value.blank? }` clause later on.

Reply:
Is it time for v3 of this PR? 😆

I prefer the idea of explicitly building the hash over the shared constant. That would require a new `#to_api_hash_with_variations` along with building the hash explicitly in `#to_api_hash`. As a consequence of that the runtime of the `brew generate-*-api` methods will probably be double what it is currently since we're doing twice the work. Building the variations is especially slow. If that sounds good, I'll close this PR and continue with that idea. Hopefully we won't need a v4.

Reply:
> `linked_keg`, `installed`, `pinned`, `outdated` will not be removed depending on the values on the generated machine (which seems undesirable).

But we know which machines we're running these commands on so that shouldn't be a problem, right?